### PR TITLE
Sonarcloud-guided clean-up

### PR DESCRIPTION
**Pull request by sjvrijn**
This PR applies a number of refactors to reduce warnings raised by Sonarcloud, and adds a few others that make the code more readable in my opinion.

Relevant issues: #5 and possibly #6

### CODE REVIEW

File: src/electricity/PretrainDataset.py [hunk 1]
```diff
@@ -22,10 +22,12 @@ def __getitem__(self, index):
...
         for i in range(len(x)):
-            prob = random.random()
-            if prob <= self.mask_prob:
+            if random.random() <= self.mask_prob:
                 prob = random.random()
-                x[i] = -1 if prob < 0.8 else np.random.normal() if prob < 0.9 else x[i]
+                if prob < 0.8:
+                    x[i] = -1
+                elif prob < 0.9:
+                    x[i] = np.random.normal()
```
Comment (OleMussmann) [inline src/electricity/PretrainDataset.py:30]: I have no idea what's happening here, but that's an issue with the original code. The refactored code is at least more readable.

File: src/electricity/config.py
Comment (OleMussmann) [inline src/electricity/config.py:1]: I'm not a big fan of either the original (too convoluted), nor the proposed code (technically correct, but is less readable and removes the option to easily change in-between numbers).

How about:

```
if args.dataset_code == "redd_lf":
    args.house_indices = [1, 2, 3, 4, 5, 6]
elif args.dataset_code == "uk_dale":
    args.house_indices = [1, 2, 3, 4, 5]
elif args.dataset_code == "refit":
    args.house_indices = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21]
else:
    raise ValueError(f"Unknown dataset code {args.dataset_code}.")
```
or
```
house_indices_per_dataset = {
        "redd_lf": [1, 2, 3, 4, 5, 6],
        "uk_dale": [1, 2, 3, 4, 5],
        "refit": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21],
        }

args.house_indices = house_indices_per_dataset[dataset_code]
```

**Review by OleMussmann — CHANGES_REQUESTED** [review]
Good changes, thank you. What do you think about the proposal for `src/electricity/config.py`?

File: src/electricity/PretrainDataset.py (hunk 1 above)
Comment (sjvrijn) [reply src/electricity/PretrainDataset.py:30]: I have no idea either why there's a double random check 🤷

File: src/electricity/config.py
Comment (sjvrijn) [reply src/electricity/config.py:1]: Fair enough, I'd prefer the 2nd implementation then, so I'll switch it to that.

**Review by OleMussmann — APPROVED** [review]
LGTM, thank you!

**Pull request merged**